#!/usr/bin/env python
PACKAGE = "velocity_controller"

from dynamic_reconfigure.parameter_generator_catkin import *

gen = ParameterGenerator()

# geometric_controller

gen.add("X", str_t, 0, "Parameter X axis", "Parameter X axis")
gen.add("UMax_x", double_t, 0, " ", 0.3, 0.0, 2.0)
gen.add("UMin_x", double_t, 0, " ", -0.3, -2.0, 0.0)

gen.add("kpx", double_t, 0, " ", 1.4, 0.0, 5.0)
gen.add("kix", double_t, 0, " ", 0.3, 0.0, 5.0)
gen.add("kdx", double_t, 0, " ", 1.1, 0.0, 5.0)


gen.add("Y", str_t, 0, "Parameter Y axis", "Parameter Y axis")
gen.add("UMax_y", double_t, 0, " ", 0.3, 0.0, 2.0)
gen.add("UMin_y", double_t, 0, " ", -0.3, -2.0, 0.0)

gen.add("kpy", double_t, 0, " ", 1.4, 0.0, 5.0)
gen.add("kiy", double_t, 0, " ", 0.3, 0.0, 5.0)
gen.add("kdy", double_t, 0, " ", 1.1, 0.0, 5.0)


gen.add("Z", str_t, 0, "Parameter Z axis", "Parameter Z axis")
gen.add("UMax_z", double_t, 0, " ", 0.5, 0.0, 2.0)
gen.add("UMin_z", double_t, 0, " ", -0.5, -2.0, 0.0)

gen.add("kpz", double_t, 0, " ", 1.8, 0.0, 5.0)
gen.add("kiz", double_t, 0, " ", 0.3, 0.0, 5.0)
gen.add("kdz", double_t, 0, " ", 3.0, 0.0, 5.0)


gen.add("UMax_yaw", double_t, 0, " ", 0.0, 0.1, 0.5)
gen.add("UMin_yaw", double_t, 0, " ", -0.3, -0.1, 0.0)

gen.add("kp_yaw", double_t, 0, " ", 1.0, 0.0, 20.0)
gen.add("ki_yaw", double_t, 0, " ", 0.0, 0.0, 20.0)
gen.add("kd_yaw", double_t, 0, " ", 0.0, 0.0, 20.0)


exit(gen.generate(PACKAGE, "velocity_controller", "VelocityController"))
